addons/loyalty/tests/test_loyalty.py

```diff
@@ -184,3 +184,36 @@ def test_prevent_archiving_product_when_archiving_program(self):
         loyalty_program.action_archive()
         # Make sure that the main product didn't get archived
         self.assertTrue(product.active)
+
+    def test_card_description_on_tag_change(self):
+        product_tag = self.env['product.tag'].create({'name': 'Multiple Products'})
+        product1 = self.env['product.product'].create({
+            'name': 'Test Product',
+            'detailed_type': 'consu',
+            'list_price': 20.0,
+            'product_tag_ids': product_tag,
+        })
+        self.env['product.product'].create({
+            'name': 'Test Product 2',
+            'detailed_type': 'consu',
+            'list_price': 30.0,
+            'product_tag_ids': product_tag,
+        })
+        reward = self.env['loyalty.reward'].create({
+            'program_id': self.program.id,
+            'reward_type': 'product',
+            'reward_product_id': product1.id,
+        })
+        reward_description_single_product = reward.description
+        reward.reward_product_tag_id = product_tag
+        reward_description_product_tag = reward.description
+        self.assertEqual(
```

Comment:
I think it would be clear if the two checks would be switched, first to check if the description is changed, then if the changed description is correct.